core/sql/generator/GenRelScan.cpp

```diff
@@ -2203,10 +2203,46 @@ short HbaseAccess::codeGen(Generator * generator)
 		       ExpTupleDesc::SQLMX_KEY_FORMAT);
 
   const ValueIdList &retColumnList = retColRefSet_; 
+  // Always get the index name -- it will be the base tablename for
+  // primary access if it is trafodion table.
+  char * tablename = NULL;
+  char * snapshotName = NULL;
+  LatestSnpSupportEnum  latestSnpSupport=  latest_snp_supported;
+  if ((getTableDesc()->getNATable()->isHbaseRowTable()) ||
+      (getTableDesc()->getNATable()->isHbaseCellTable()))
+    {
+      tablename =
+        space->AllocateAndCopyToAlignedSpace(
+                                             GenGetQualifiedName(getTableName().getQualifiedNameObj().getObjectName()), 0);
+      latestSnpSupport = latest_snp_not_trafodion_table;
+    }
+  else
+    {
+      if (getIndexDesc() && getIndexDesc()->getNAFileSet())
+      {
+         tablename = space->AllocateAndCopyToAlignedSpace(GenGetQualifiedName(getIndexDesc()->getNAFileSet()->getFileSetName()), 0);
+         if (getIndexDesc()->isClusteringIndex())
+         {
+            //base table
+            snapshotName = (char*)getTableDesc()->getNATable()->getSnapshotName() ;
+           if (snapshotName == NULL)
+             latestSnpSupport = latest_snp_no_snapshot_available;
+          }
+          else
+            latestSnpSupport = latest_snp_index_table;
```

Comment:
This part of the code is moved here from few lines below in the same function just to get the tablename. The snapshot functionality is not affected by this change. Yes. We need to figure out  the consequences if the snapshot for the index is not created, when table has snapshot and vice versa. But, it is outside the scope of this change